### PR TITLE
Foundations for styling

### PR DESCRIPTION
**Pull request by nath-green**
Creation of files and folders to create a base for styling going forward.

PR includes:
- Resets
- Empty shell for `atoms` and `molecules`
- Ported colors
- Helper class for `u-background-${color}`
- Helper functions for lighten and darken
- Typographic scale naming with golden ratio line height function
- `font`, `font-size` and `line-height` functions (using `map-deep-get`)
- Tidy of grid
- Spacing variable and helper classes rename
- Basic global body styling

### CODE REVIEW

File: src/styles/settings/_colors.scss [hunk 1]
```diff
@@ -1,23 +1,47 @@
...
   secondary: #5E86CB,
-  pavement: #d5d5d7,
+  pavement: #D5D5D7,
   cloud: #F9F9F9,
   ash: #B3B3B5,
   charcoal: #222630,
-  success: #27ce5d,
+  success: #27CE5D,
   error: #FF3D3D,
   coral: #FF6B6B
```
Comment (aorv) [inline src/styles/settings/_colors.scss]: 👌

File: src/styles/settings/_colors.scss [hunk 2]
```diff
@@ -1,23 +1,47 @@
...
+
+@function darken-color($key) {
+  @if map-has-key($colors, $key) {
+    @return lighten(map-get($colors, $key), $darken-pc);
+  }
+}
+
 @each $name, $color in $colors {
   .u-text-#{$name} {
     color: color($name)!important;
```
Comment (aorv) [inline src/styles/settings/_colors.scss]: Do you think the `!importants` will be required?

File: src/styles/settings/_colors.scss [hunk 3]
```diff
@@ -1,23 +1,47 @@
...
 }
+
+@each $name, $color in $colors {
+  .u-background-#{$name} {
+    background-color: color($name)!important;
+  }
+}
+
+$body-color: map-get($colors, charcoal);
+$body-background-color: map-get($colors, cloud);
```
Comment (aorv) [inline src/styles/settings/_colors.scss]: Could use the `color` function for these 2 vars

File: src/styles/settings/_colors.scss (hunk 2 above)
Comment (aorv) [inline src/styles/settings/_colors.scss]: Do you think the !importants will be required?

Comment (nath-green) [reply src/styles/settings/_colors.scss]: Yeah man, utility classes should have `!important` on them as they should outrank everything. Just looked at Foundation and Bootstrap and all their utility classes have it too 👍

Comment (aorv) [reply src/styles/settings/_colors.scss]: roger roger

**Pull request merged**